### PR TITLE
Ensure OTLP receiver handles consume errors correctly [OTLP/HTTP]

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** Follow the receiver contract and return 503 for non-permanent and 400 for permanent errors for OTLP/HTTP receiver.

Leave the "Retry-After" field blank and let the client implement an exponential backoff strategy.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/4335

### CODE REVIEW

**Comment by atoulme** [comment]
Please add tests and changelog.

**Review by jpkrohling — APPROVED** [review]
LGTM, but @astencel-sumo might want to take a look as well. We had some discussions about this as part of #7486

**Comment by VihasMakwana** [comment]
**This PR depends on https://github.com/open-telemetry/opentelemetry-collector/pull/8080. Marking as draft for now**

**Comment by jpkrohling** [comment]
ping @astencel-sumo

File: receiver/otlpreceiver/otlp_test.go [hunk 1]
```diff
@@ -954,6 +955,62 @@ func TestHTTPMaxRequestBodySize_TooLarge(t *testing.T) {
 	testHTTPMaxRequestBodySizeJSON(t, traceJSON, len(traceJSON)-1, 400)
 }
 
+func TestHTTPErrors_PermanentAndNonPermanent(t *testing.T) {
+	errorTypes := []struct {
+		permanent          bool
+		nonPermanent       bool
```
Comment (evan-bradley) [inline receiver/otlpreceiver/otlp_test.go:961]: It looks like these fields are unused, can they be removed?

**Pull request closed without merging**